### ISSUE TITLE
Failed to accept: permission denied

### ISSUE THREAD

**Issue by DimStar77** [openSUSE/open-build-service#9310]
Last night, I was running a 'staging accept' for openSUSE:Factory. Staging:adi:21 containes some new packages (from a new devel project) coming to Factory.

While accepting those requests, I was only given a 'error 403' - The problem is that Factory has 'OBS:InitializeDevelProject' defined - but my user does not (well, did not - fixed by now together with a maintainer) maintainer access in multimedia:voice-assistant.

There needs to be a better permission model for this case (of OBS initializing a devel package, but the acceptor only having rights in the target project, but not in the source project)

**Comment by hennevogel** [comment]
`OBS:InitializeDevelProject` only changes things in the target project (openSUSE:Factory). So this must be something else.

**Comment by adrianschroeter** [comment]
add a devel reference in the target project requires maintainer rights in the source.

**Comment by DimStar77** [comment]
IIRC, it also converts the source to a link to the target

**Comment by adrianschroeter** [comment]
yes, it does. both actions where you need write access in source. Or you could misuse this feature.

**Comment by hennevogel** [comment]
> add a devel reference in the target project requires maintainer rights in the source on purpose.

So how do you imagine a "better permission model for this case"?

**Comment by DimStar77** [comment]
> So how do you imagine a "better permission model for this case"?

Probably we need to try to attack it from a different angle than OBS/permissions then. The staging project should not enter the 'accept' route if I don't have permission to handle the accept of all requests in the staging

**Comment by DimStar77** [comment]
(a first step we can probably easily do is for one of the factory bots to decline requests coming from a devel prj where factory-maintainers is not granted maintainer access)

**Comment by hennevogel** [comment]
We can block merging in this case too and throw an error.
